File: 219-contains-duplicate-ii/contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k)
    {
        set<int> window;
        int l = 0;

        for (int r = 0; r < nums.size(); r++)
        {
            if (r - l > k)
            {
                window.erase(nums[l]);
                l++;
            }
            auto it = window.find(nums[r]);
            if (it != window.end())
                return true;
            window.insert(nums[r]);
        }
        return false;
    }
};

File: 219-contains-duplicate-ii/contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        m = defaultdict(int)
        j = 0

        for i, num in enumerate(nums):
            if i - j > k:
                del m[nums[j]]
                j += 1
            if nums[i] in m:
                return True
            m[nums[i]] += 1
            i += 1
        return False


                
        